### ISSUE TITLE
showMenu() can only show vertical menu

### ISSUE THREAD

**Issue by rxlabz** [flutter/flutter#10382]
I maybe wrong but there is actually no easy way to display a grid or horizontal menu with the showMenu. Would be great if we could choose the layout of the PopupMenuItems.

**Comment by rxlabz** [comment]
it's not an option defined in Material Design guidelines, so I guess it's not an "issue"

**Comment by Hixie** [comment]
You can create a new PopupMenuEntry subclass that does whatever you want. So long as you implement `height` to accurately describe the height of your widget, you can make it do whatever. Just call `Navigator.pop` when you want the menu to close.

**Comment by Hixie** [comment]
I'll make PopupMenuEntry more useful for this purpose and improve the docs.

**Comment by rxlabz** [comment]
I've studied more attentively the showMenu() method and all the classes it uses, and finally successfully created a *showGridMenu()* , which basically uses a GridView as a child instead the  ListBody.  
In my first attempt, the animation part lost me a little, I found it difficult to understand and adapt, but once I removed it, things became clearer. Now that I have the correct static display, I'll try to animate it again.
Anyway, once you get it, this PopupRoute system seems great and, indeed, give a lot of options. Thanks !
